src/index.js

```diff
@@ -1,567 +1,363 @@
- * @return {Array of strings} resulting shards
- */
-const shardHex = hex => {
-  const buffer = hex2buf(hex);
-  const sharded = shardBuffer(buffer);
-  const shards = sharded.map(pair =>
-    lodash.reduce(pair, (acc, arr) =>
-      acc + buf2hex(Buffer.from(arr)), ''));
-
-  if (shardsConsistent(combineHex, hex, lodash.isEqual, shards)) {
-    return shards;
-  } else {
-    // shards must always be consistent, so this should be unreachable
-    /* istanbul ignore next */
-    throw("shardHex: inconsistent shards -- please report this as a bug");
   }
 }
 
 
 
 /**
- * Encode a @q-encoded string as three shards, such that any two shards can be
- * combined to recover it.
- *
- * @param  {string}  string @q-encoded string
- * @return {Array of strings} resulting shards
+ * Convert a hex-encoded secp256k1 public key into an Ethereum address.
+ * @param  {String}  pub a (compressed) hex-encoded public key
+ * @return  {String}  the corresponding Ethereum address
  */
-const shardPatq = patq => {
-  const hexed = shardHex(ob.patq2hex(patq))
-  const shards = hexed.map(ob.hex2patq)
-
-  if (shardsConsistent(combinePatq, patq, ob.eqPatq, shards)) {
-    return shards;
-  } else {
-    // shards must always be consistent, so this should be unreachable
-    /* istanbul ignore next */
-    throw("shardPatq: inconsistent shards -- please report this as a bug");
-  }
+const addressFromSecp256k1Public = pub => {
+  const hashed = util.keccak256(Buffer.from(pub, 'hex'))
+  const addr = util.addHexPrefix(hashed.slice(12).toString('hex'))
+  return util.toChecksumAddress(addr)
 }
 
 
 
 /**
- * Produce three shards from a buffer such that any two of them can be used to
- * reconstruct it.
- * @param  {Buffer}  buffer arbitrary buffer
- * @return {Array of Array of integers} sharded buffer
+ * Convert a hex-encoded secp256k1 private key into an Ethereum address.
+ * @param  {String}  pub a hex-encoded private key
+ * @return  {String}  the corresponding Ethereum address
  */
-const shardBuffer = buffer => {
-  const r1 = crypto.getRandomValues(new Uint8Array(buffer.length));
-  const r2 = crypto.getRandomValues(new Uint8Array(buffer.length));
-
-  const k  = Array.from(buffer);
-  const k1 = Array.from(r1);
-  const k2 = Array.from(r2);
-
-  const k0 = reduceByXor([k, k1, k2]);
-
-  const anyEqual =
-       lodash.isEqual(k0, k1)
-    || lodash.isEqual(k0, k2)
-    || lodash.isEqual(k1, k2)
-
-  return anyEqual
-    ? shardBuffer(buffer)
-    : [[k0, k1], [k0, k2], [k1, k2]];
+const addressFromSecp256k1Private = priv => {
+  const pub = util.secp256k1.publicKeyCreate(Buffer.from(priv, 'hex'))
+  return addressFromSecp256k1Public(pub)
 }
 
 
 
 /**
- * Combine pieces of a sharded buffer together to recover the original buffer.
- * @param  {Array of Array of integers}  shards a collection of shards
- * @return {Buffer} the unsharded buffer
+ * Convert a hex-encoded Ed25519-variant Urbit public network key into an
+ * Ethereum address.
```

Comment:
Lol why. This is neat, but doesn't actually need to happen. Urbit keys are for Urbit-use only, representing seed, pub and prv as hex strings is totally fine here.

Reply:
Yeah, I thought.. do we actually need this?  And then erred on the side of caution.. 😁   Will strip out.